File: paper_artifacts/experiments/llr8/artifacts/sources/oss120b-c/tsvc_2_s3111__submitted.c
#include <stddef.h>
#include <stdint.h>

void tsvc_2_s3111_fp64(double *restrict a, double *restrict b, int64_t LEN_1D, uint8_t *workspace, int64_t workspace_bytes) {
    double sum_val = 0.0;
    #pragma omp simd reduction(+:sum_val)
    for (int64_t i = 0; i < LEN_1D; ++i) {
        double ai = a[i];
        if (ai > 0.0) {
            sum_val += ai;
        }
    }
    b[0] = sum_val;
    (void)workspace;
    (void)workspace_bytes;
}

File: paper_artifacts/experiments/llr9/artifacts/sources/qwen38-fortran-skills/tsvc_2_s3111__submitted.f90
subroutine tsvc_2_s3111_fp64(a, b, len_1d, workspace, workspace_size) bind(C)
  use iso_c_binding
  implicit none
  integer(c_int64_t), value, intent(in) :: len_1d
  type(c_ptr), intent(in) :: workspace
  integer(c_int64_t), value, intent(in) :: workspace_size
  real(c_double), intent(in)    :: a(len_1d)
  real(c_double), intent(out)   :: b(2)

  integer(c_int64_t) :: i, n
  real(c_double) :: s

  n = len_1d
  s = 0.0d0
  !$omp parallel do simd reduction(+:s)
  do i = 1, n
    s = s + merge(a(i), 0.0d0, a(i) > 0.0d0)
  end do
  b(1) = s
end subroutine tsvc_2_s3111_fp64
